addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -29,6 +29,35 @@ class StockPickingBatch(models.Model):
         help="Priority of a batch is the maximum priority of its pickings."
     )
 
+    state = fields.Selection(
+        selection_add=[
+            ('waiting', 'Waiting'),
+            ('ready', 'Ready'),
+        ],
+        compute='_compute_state',
+        store=True,
+    )
+
+    @api.multi
+    def confirm_picking(self):
+        """Over write method to use our waiting state and recompute state
+           incase this should be ready or in_progress
+        """
+        pickings_todo = self.mapped('picking_ids')
+        self.write({'state': 'waiting'})  # Get it out of draft
```

Comment:
Do we want to enforce that the we're doing a `draft` --> `waiting` transition? 

If so, is such condition expected (so we can assert here - i.e. it's up to the caller to ensure that)? Or do we want to raise an invalid state error?

Reply:
The `draft` -> `waiting` transition is a forced one its just them saying I want to do this piece of work.
And setting it as `waiting` is purely a lets get started on our automatic transitions path kinda thing.